### PR TITLE
Bugfix: Fixed the collapse button issue for needs.

### PR DESCRIPTION
**Pull request by iSOLveIT**
Fixed the collapse button issue for needs which also affected nested needs. This fixes issue #659.

### CODE REVIEW

File: sphinx_needs/css/blank/styles.css [hunk 1]
```diff
@@ -90,3 +90,6 @@ table.needs_style_blue_bar {
     background: linear-gradient(to right, transparent 90%, #0fbcf9 10%);
 }
 
+.is_visible{
```
Comment (ubmarco) [inline sphinx_needs/css/blank/styles.css:93]: This is quite a generic class selector that might conflict with custom end-user CSS.
Can you make it more specific to a need/table/header/... or change the name to something like 'collape_is_visible'?
Also the name is inverted somehow.

File: sphinx_needs/css/dark/styles.css [hunk 2]
```diff
@@ -115,3 +115,7 @@ table.needs_style_yellow_bar {
 table.needs_style_blue_bar {
     background: linear-gradient(to right, transparent 90%, #0fbcf9 10%);
 }
+
+.is_visible{
```
Comment (ubmarco) [inline sphinx_needs/css/dark/styles.css:119]: same here

File: sphinx_needs/css/modern/styles.css [hunk 3]
```diff
@@ -124,3 +124,6 @@ table.needs_style_yellow_bar {
 table.needs_style_blue_bar {
     background: linear-gradient(to right, transparent 90%, #0fbcf9 10%);
 }
+.is_visible{
```
Comment (ubmarco) [inline sphinx_needs/css/modern/styles.css:127]: same here

**Review by ubmarco — CHANGES_REQUESTED** [review]

File: sphinx_needs/css/blank/styles.css (hunk 1 above)
Comment (iSOLveIT) [reply sphinx_needs/css/blank/styles.css]: Changes implemented in the new commit.

File: sphinx_needs/css/dark/styles.css (hunk 2 above)
Comment (iSOLveIT) [reply sphinx_needs/css/dark/styles.css]: Changes implemented in the new commit.

File: sphinx_needs/css/blank/styles.css [hunk 4]
```diff
@@ -90,3 +90,6 @@ table.needs_style_blue_bar {
     background: linear-gradient(to right, transparent 90%, #0fbcf9 10%);
 }
 
+.collapse_is_visible {
```
Comment (ubmarco) [inline sphinx_needs/css/blank/styles.css:93]: I just compared this class with below jQuery selector. Would the selector here still work if prefixed like this
```
span.needs.collapse_is_visible {
```
? If yes, I'd appreciate to use a more detailed selector, just to be sure it does not clash with user-space CSS.

**Review by ubmarco — APPROVED** [review]
Thanks for the quick fix, this is working as intended.
Thanks also for the other small fixes (typos).

File: sphinx_needs/css/blank/styles.css (hunk 4 above)
Comment (iSOLveIT) [reply sphinx_needs/css/blank/styles.css:93]: Yes it will.

Comment (ubmarco) [reply sphinx_needs/css/blank/styles.css:93]: Ok, thanks for the update.
Then I'd vote for
```
span.needs.collapse_is_hidden {
    display: none;
} 
```
The selector is as specific as possible.
A class that makes something disappear should have something like 'hidden' in its name.
Any thoughts?

Comment (iSOLveIT) [reply sphinx_needs/css/blank/styles.css:93]: Yeah, collapse-is-hidden is ok too

**Pull request merged**